### ISSUE TITLE
'Problem with constructing exception message' error when executing compileKotlin task with gradle

### ISSUE THREAD

**Issue by octaviospain** [Kotlin/kotlinx.serialization#2424]
I have a gradle multi module kotlin project that uses the built-in kotlinx serialization package. 
After some refactor, trying to build it results in a very poorly descriptive error message that I cannot debug and trace back to the problem:

```
exception: exception in thread "main" java.lang.IllegalStateException: Problem with constructing exception message
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializerClassLowering.lower(SerializationLoweringExtension.kt:228)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitClass(SerializationLoweringExtension.kt:49)
exception:      at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitClass(IrElementVisitorVoid.kt:111)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitClass(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitClass(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlin.ir.declarations.IrClass.accept(IrClass.kt:73)
exception:      at org.jetbrains.kotlin.ir.declarations.IrFile.acceptChildren(IrFile.kt:37)
exception:      at org.jetbrains.kotlin.ir.visitors.IrVisitorsKt.acceptChildrenVoid(IrVisitors.kt:15)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitElement(SerializationLoweringExtension.kt:45)
exception:      at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitPackageFragment(IrElementVisitorVoid.kt:190)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitPackageFragment(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitFile(IrElementVisitorVoid.kt:200)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitFile(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitFile(IrElementVisitorVoid.kt:198)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitFile(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt$runOnFileInOrder$1.visitFile(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlin.ir.declarations.IrFile.accept(IrFile.kt:31)
exception:      at org.jetbrains.kotlin.ir.visitors.IrVisitorsKt.acceptVoid(IrVisitors.kt:11)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtensionKt.runOnFileInOrder(SerializationLoweringExtension.kt:43)
exception:      at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationLoweringExtension.generate(SerializationLoweringExtension.kt:193)
exception:      at org.jetbrains.kotlin.backend.jvm.JvmIrCodegenFactory.convertToIr$lambda$1(JvmIrCodegenFactory.kt:222)
exception:      at org.jetbrains.kotlin.psi2ir.Psi2IrTranslator.generateModuleFragment(Psi2IrTranslator.kt:107)
exception:      at org.jetbrains.kotlin.backend.jvm.JvmIrCodegenFactory.convertToIr(JvmIrCodegenFactory.kt:255)
exception:      at org.jetbrains.kotlin.backend.jvm.JvmIrCodegenFactory.convertToIr(JvmIrCodegenFactory.kt:59)
exception:      at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.convertToIr(KotlinToJVMBytecodeCompiler.kt:224)
exception:      at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:101)
exception:      at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:47)
exception:      at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:168)
exception:      at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:53)
exception:      at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:100)
exception:      at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:46)
exception:      at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
exception:      at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:79)
exception:      at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:43)
exception:      at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:180)
exception:      at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit$default(CLITool.kt:178)
exception:      at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMain(CLITool.kt:167)
exception:      at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler$Companion.main(K2JVMCompiler.kt:251)
exception:      at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.main(K2JVMCompiler.kt)
```

**Environment**
 - Kotlin version: `1.9.0`
 - Library version: `1.5.1`
 - Kotlin platforms: `JVM`
 - Gradle version: `8.3`
 - OS: `Linux 6.1.44-1-MANJARO amd64`
 - Java: `17.0.5 (BellSoft 17.0.5+8-LTS)`
 
 I'm attaching the full output of the command `gradle build --stacktrace` in `build.log`, and the created `errors-1693134472612.log`
 
[errors-1693134472612.log](https://github.com/Kotlin/kotlinx.serialization/files/12447845/errors-1693134472612.log)
[build.log](https://github.com/Kotlin/kotlinx.serialization/files/12447846/build.log)

**Comment by sandwwraith** [comment]
Unfortunately I can't do anything meaningful without the project source code triggering the error (because message is, well, unavailable). Do you have minimized reproducer by any chance?

**Comment by pdvrieze** [comment]
@octaviospain Does the problem persist when you first clean the project and do a full rebuild? (I have encountered various issues with incremental builds and the serialization plugin)

**Comment by octaviospain** [comment]
I imagined someone would ask at some point. Unfortunately I didn't post source code because it was quite difficult to reproduce the same behavior that ocured during a refactor.
At the end I opted for rethink the serialization strategy I used and I'm not facing this problem any more.

However, is not the first time that I get this poorly descriptive exception messages from the library when something is wrong with the serialization.
